config/routes.rb

```diff
@@ -28,6 +28,12 @@
     resources  :categories, only: :index
     resources  :prospective_users, only: :create
     resources  :assessments, only: [:index, :show, :update, :create] do
+      collection do
+        resources :shared, controller: 'shared_assessments', only: [:show], param: :token do
```

Comment:
The fact that we're not using generated routes doesn't give me enough reasons to change this over full paths. It's unclear to me what's the advantage of your proposed solution.